framework/src/org/apache/cordova/CordovaPlugin.java

```diff
@@ -359,4 +361,38 @@ public boolean onReceivedClientCertRequest(CordovaWebView view, ICordovaClientCe
      */
     public void onConfigurationChanged(Configuration newConfig) {
     }
+
+    /**
+     * Called by the Plugin Manager when we need to actually request permissions
+     *
+     * @return              Returns the permission that was stored in the plugin
+     */
+
+    public String[] getPermissionRequest() {
+        return permissions;
+    }
+
+    public boolean hasPermisssion() {
+        for(String p : permissions)
+        {
+            if(PackageManager.PERMISSION_DENIED == cordova.getActivity().checkSelfPermission(p))
```

Comment:
The geolocation plugin ONLY handles the permissions, and since we require both the coarse location and the fine location to be requested, we send them both to the activity.  There isn't this sort of behaviour in any other plugin that we currently maintain.  However, it's possible that other plugins will want to send a bunch of permission as a batch, so we facilitate that here, which is why it's public.

That said, we don't do this for the media plugin.  I chose the one permission at a time approach because it's easier to control a plugin's callbackContext that way.